nuxeo-multi-tenant-core/src/test/java/org/nuxeo/ecm/multi/tenant/TestMultiTenantDirectories.java

```diff
@@ -0,0 +1,161 @@
+import org.nuxeo.directory.test.DirectoryFeature;
+import org.nuxeo.ecm.core.api.CloseableCoreSession;
+import org.nuxeo.ecm.core.api.DocumentModelList;
+import org.nuxeo.ecm.core.api.NuxeoPrincipal;
+import org.nuxeo.ecm.core.test.CoreFeature;
+import org.nuxeo.ecm.core.test.annotations.Granularity;
+import org.nuxeo.ecm.core.test.annotations.RepositoryConfig;
+import org.nuxeo.ecm.directory.DirectoryException;
+import org.nuxeo.ecm.directory.Session;
+import org.nuxeo.ecm.directory.api.DirectoryService;
+import org.nuxeo.ecm.platform.test.PlatformFeature;
+import org.nuxeo.ecm.platform.usermanager.UserManager;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+
+/**
+ * Test class for the management of directories with multitenancy.
+ *
+ * @since 10.2
+ */
+@RunWith(FeaturesRunner.class)
+@Features({ DirectoryFeature.class, PlatformFeature.class })
+@Deploy("org.nuxeo.ecm.default.config")
+@Deploy("org.nuxeo.ecm.platform.userworkspace.core")
+@Deploy("org.nuxeo.ecm.multi.tenant")
+@Deploy("org.nuxeo.ecm.multi.tenant:multi-tenant-test-contrib.xml")
+@Deploy("org.nuxeo.ecm.multi.tenant:multi-tenant-enabled-default-test-contrib.xml")
+@Deploy("org.nuxeo.ecm.multi.tenant:multi-tenant-test-directories-contrib.xml")
+@Deploy("org.nuxeo.ecm.platform.test:test-usermanagerimpl/userservice-config.xml")
+@RepositoryConfig(cleanup = Granularity.METHOD, init = MultiTenantRepositoryInit.class)
+public class TestMultiTenantDirectories {
+
+    @Inject
+    protected CoreFeature coreFeature;
+
+    @Inject
+    protected DirectoryService directoryService;
+
+    @Inject
+    protected UserManager userManager;
+
+    @Test
+    public void iCanCreateSameDirectoryEntryInDifferentTenant() throws LoginException {
+        // Add user0 and user1 to the powerusers group so they can edit the directory
+        Framework.doPrivileged(() -> {
+            NuxeoPrincipal user0Principal = userManager.getPrincipal("user0");
+            List<String> groups = user0Principal.getGroups();
+            groups.add(Constants.POWER_USERS_GROUP);
+            user0Principal.setGroups(groups);
+            userManager.updateUser(user0Principal.getModel());
+            NuxeoPrincipal user1Principal = userManager.getPrincipal("user1");
+            groups = user1Principal.getGroups();
+            groups.add(Constants.POWER_USERS_GROUP);
+            user1Principal.setGroups(groups);
+            userManager.updateUser(user1Principal.getModel());
+        });
+
+        // Map of the entry to create per tenant
+        Map<String, Object> newEntry = new HashMap<>();
+        newEntry.put("id", "key0");
+        newEntry.put("label", "Label Key 0");
+
+        // As user0, add a new entry in directory
+        LoginContext loginContext = Framework.loginAsUser("user0");
+        try (CloseableCoreSession userSession = openSession()) {
```

Comment:
Do we need to open a `CoreSession` ? other tests like `TestDirectory#testCreateEntry` just open the directory session.

Reply:
I need to create the entry as user0 or user1. Otherwise, the tenantId will not be fetched at the entry creation and I won't be able test the behavior I want.